src/routes/Profile/EducationHistory/modals/AddOrEditKoulutusModal.tsx

```diff
@@ -151,28 +158,39 @@ const EditKoulutusModal = ({ isOpen, onClose, koulutuskokonaisuusId: id, koulutu
         .refine((data) => !data.loppuPvm || data.alkuPvm <= data.loppuPvm),
     ),
     defaultValues: async () => {
-      const [{ data: osaamiset }, { data: koulutus }] = await Promise.all([
-        client.GET('/api/profiili/osaamiset'),
-        client.GET(KOULUTUS_API_PATH, {
-          params: {
-            path: { id, koulutusId },
-          },
-        }),
-      ]);
+      if (koulutusId) {
+        const [{ data: osaamiset }, { data: koulutus }] = await Promise.all([
+          client.GET('/api/profiili/osaamiset'),
+          client.GET(`${KOULUTUKSET_API_PATH}/{koulutusId}`, {
+            params: {
+              path: { id, koulutusId },
+            },
+          }),
+        ]);
 
-      return {
-        id: koulutus?.id ?? '',
-        nimi: koulutus?.nimi?.[language] ?? '',
-        kuvaus: koulutus?.kuvaus?.[language] ?? '',
-        alkuPvm: koulutus?.alkuPvm ?? '',
-        loppuPvm: koulutus?.loppuPvm ?? '',
-        osaamiset:
-          koulutus?.osaamiset?.map((osaaminenId) => ({
-            id: osaaminenId,
-            nimi: osaamiset?.find((o) => o.osaaminen?.uri === osaaminenId)?.osaaminen?.nimi ?? {},
-            kuvaus: osaamiset?.find((o) => o.osaaminen?.uri === osaaminenId)?.osaaminen?.kuvaus ?? {},
-          })) ?? [],
-      };
+        return {
+          id: koulutus?.id ?? '',
+          nimi: koulutus?.nimi?.[language] ?? '',
```

Comment:
You can use `nimi: getLocalizedText(koulutus?.nimi)` from utils. No need for `language` or `?? ''`

Reply:
I didn't change that part. Better to refactor it in its own ticket.